### PR TITLE
add a prometheus record to monitor the state of enmasse-operator

### PR DESCRIPTION
**Pull request by jbtrystram**
### Type of change

- Enhancement / new feature

### CODE REVIEW

File: controller-manager/src/main/resources/templates/PrometheusRules-enmasse.yaml [hunk 1]
```diff
@@ -20,6 +20,9 @@ spec:
     - record: enmasse_component_health
       expr: up{job="address-space-controller"} or on(namespace) (1- absent(up{job="address-space-controller"}) )
 
+    - record: enmasse_component_health
+        expr: up{job="enmasse-operator"} or on(namespace) (1- absent(up{job="enmasse-operator"}) )
```
Comment (robshelly) [inline controller-manager/src/main/resources/templates/PrometheusRules-enmasse.yaml]: The operator is actually exporting metrics with the label `job="enmasse-operator-metrics"`.

File: controller-manager/src/main/resources/templates/PrometheusRules-enmasse.yaml [hunk 2]
```diff
@@ -20,6 +20,9 @@ spec:
     - record: enmasse_component_health
       expr: up{job="address-space-controller"} or on(namespace) (1- absent(up{job="address-space-controller"}) )
 
+    - record: enmasse_component_health
+        expr: up{job="job="enmasse-operator-metrics""} or on(namespace) (1- absent(up{job="job="enmasse-operator-metrics""}) )
```
Comment (robshelly) [inline controller-manager/src/main/resources/templates/PrometheusRules-enmasse.yaml]: Test this out and discovered there's an extra tab at the start of the line which is causing the creation of the resource to fail

Comment (jbtrystram) [reply controller-manager/src/main/resources/templates/PrometheusRules-enmasse.yaml]: @robshelly indeed. I updated

File: controller-manager/src/main/resources/templates/PrometheusRules-enmasse.yaml [hunk 3]
```diff
@@ -20,6 +20,9 @@ spec:
     - record: enmasse_component_health
       expr: up{job="address-space-controller"} or on(namespace) (1- absent(up{job="address-space-controller"}) )
 
+    - record: enmasse_component_health
+      expr: up{job="job="enmasse-operator-metrics""} or on(namespace) (1- absent(up{job="job="enmasse-operator-metrics""}) )
```
Comment (robshelly) [inline controller-manager/src/main/resources/templates/PrometheusRules-enmasse.yaml]: My bad! I thought I was the indentation causing the error but I just noticed there's another typo. `job="` is in the labels twice

**Pull request merged**